### PR TITLE
Avoiding rebuilds by use of configure_file()

### PR DESCRIPTION
**Pull request by borisfom**
This fixes the issue of library rebuilding all the time.

### CODE REVIEW

**Comment by abergeron** [comment]
This doesn't work because we need the abi_version header available for inclusion at build time.

**Comment by borisfom** [comment]
Why, it worked for me:
bfomitchev@borisf-desktop:~/git/libgpuarray$ rm -fr build
bfomitchev@borisf-desktop:~/git/libgpuarray$ mkdir build && cd build
bfomitchev@borisf-desktop:~/git/libgpuarray/build$ cmake -DCMAKE_INSTALL_PREFIX=../install ..
-- The C compiler identification is GNU 4.8.4
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Looking for strlcat
-- Looking for strlcat - not found
-- Looking for mkstemp
-- Looking for mkstemp - found
-- Found PkgConfig: /usr/bin/pkg-config (found version "0.26")
-- checking for one of the modules 'check'
Tests disabled because Check was not found
-- Configuring done
-- Generating done
-- Build files have been written to: /home/bfomitchev/git/libgpuarray/build
bfomitchev@borisf-desktop:~/git/libgpuarray/build$ find . -name abi_version.h
./src/gpuarray/abi_version.h
bfomitchev@borisf-desktop:~/git/libgpuarray/build$ make install
Scanning dependencies of target gpuarray
[  1%] Building C object src/CMakeFiles/gpuarray.dir/cache/lru.c.o
[  2%] Building C object src/CMakeFiles/gpuarray.dir/cache/twoq.c.o
...
-- Installing: /home/bfomitchev/git/libgpuarray/install/include/gpuarray/abi_version.h
-- Up-to-date: /home/bfomitchev/git/libgpuarray/install/include/gpuarray/config.h
.. successful completion ...

**Comment by abergeron** [comment]
It fails on travis.  I'll try it out locally tomorrow and if it works for me I'll try to debug travis.

**Comment by abergeron** [comment]
Apparently it works if you have a previous install since it finds the installed abi_version.h.

On travis there are no previous installs so it fails.

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -102,9 +102,7 @@ string(REPLACE "." ";" GPUARRAY_ABI_VERSION_NUMBERS ${GPUARRAY_ABI_VERSION})
 list(GET GPUARRAY_ABI_VERSION_NUMBERS 0 GPUARRAY_ABI_VERSION_MAJOR)
 list(GET GPUARRAY_ABI_VERSION_NUMBERS 1 GPUARRAY_ABI_VERSION_MINOR)
 math(EXPR GPUARRAY_ABI_NUMBER "1000*${GPUARRAY_ABI_VERSION_MAJOR} + ${GPUARRAY_ABI_VERSION_MINOR}")
-FILE(WRITE gpuarray/abi_version.h
-"\#ifndef GPUARRAY_ABI_VERSION\n\#define GPUARRAY_ABI_VERSION ${GPUARRAY_ABI_NUMBER}\n\#endif\n"
-)
+configure_file(gpuarray/abi_version.h.in gpuarray/abi_version.h @ONLY)
```
Comment (abergeron) [inline src/CMakeLists.txt]: You can prefix the file names with ${CMAKE_CURRENT_SOURCE_DIR} and it will place the file in the right place.

**Comment by abergeron** [comment]
This was fixed in another PR.

**Pull request closed without merging**